modules/gapi/misc/python/python_bridge.hpp

```diff
@@ -190,6 +192,134 @@ class GAPI_EXPORTS_W_SIMPLE GArrayT
     Storage m_arg;
 };
 
+namespace gapi {
+
+class GAPI_EXPORTS_W_SIMPLE GOutputs
+{
+public:
+    GOutputs() = default;
+    GOutputs(const std::string& id, cv::GKernel::M outMeta, cv::GArgs &&ins);
+
+    GAPI_WRAP cv::GMat     getGMat();
+    GAPI_WRAP cv::GScalar  getGScalar();
+    GAPI_WRAP cv::GArrayT  getGArray(cv::gapi::ArgType type);
+    GAPI_WRAP cv::GOpaqueT getGOpaque(cv::gapi::ArgType type);
+
+private:
+    class Priv;
+    std::shared_ptr<Priv> m_priv;
+};
+
+GOutputs op(const std::string& id, cv::GKernel::M outMeta, cv::GArgs&& args);
+
+template <typename... T>
+GOutputs op(const std::string& id, cv::GKernel::M outMeta, T&&... args)
+{
+    return op(id, outMeta, cv::GArgs{cv::GArg(std::forward<T>(args))... });
+}
+
+} // namespace gapi
 } // namespace cv
 
+cv::gapi::GOutputs cv::gapi::op(const std::string& id,
+                                cv::GKernel::M outMeta,
+                                cv::GArgs&& args)
+{
+    cv::gapi::GOutputs outputs{id, outMeta, std::move(args)};
+    return outputs;
+}
+
+class cv::gapi::GOutputs::Priv
+{
+public:
+    Priv(const std::string& id, cv::GKernel::M outMeta, cv::GArgs &&ins);
+
+    cv::GMat     getGMat();
+    cv::GScalar  getGScalar();
+    cv::GArrayT  getGArray(cv::gapi::ArgType);
+    cv::GOpaqueT getGOpaque(cv::gapi::ArgType);
+
+private:
+    int output = 0;
+    std::unique_ptr<cv::GCall> m_call;
+};
+
+cv::gapi::GOutputs::Priv::Priv(const std::string& id, cv::GKernel::M outMeta, cv::GArgs &&args)
```

Comment:
So I believe it is the same problem with _not_ having a .cpp file isn't it?

As the file gets larger and larger. I believe keeping details like this off the header becomes a problem

Reply:
Ok,  as option it can `python_bridge_impl.hpp` which will be included to `python_pridge.hpp`

Reply:
i'd vote for that if it works